awx/main/tasks/system.py

```diff
@@ -0,0 +1,915 @@
+# system.py
+from collections import namedtuple
+import functools
+import importlib
+import json
+import logging
+import os
+from io import StringIO
+from contextlib import redirect_stdout
+import shutil
+import time
+from distutils.version import LooseVersion as Version
+
+# Django
+from django.conf import settings
+from django.db import transaction, DatabaseError, IntegrityError
+from django.db.models.fields.related import ForeignKey
+from django.utils.timezone import now
+from django.utils.encoding import smart_str
+from django.contrib.auth.models import User
+from django.utils.translation import ugettext_lazy as _
+from django.utils.translation import gettext_noop
+from django.core.cache import cache
+from django.core.exceptions import ObjectDoesNotExist
+
+# Django-CRUM
+from crum import impersonate
+
+
+# Runner
+import ansible_runner
+import ansible_runner.cleanup
+
+# dateutil
+from dateutil.parser import parse as parse_date
+
+# AWX
+from awx import __version__ as awx_application_version
+from awx.main.access import access_registry
+from awx.main.models import (
+    Schedule,
+    TowerScheduleState,
+    Instance,
+    InstanceGroup,
+    UnifiedJob,
+    Notification,
+    Inventory,
+    SmartInventoryMembership,
+    Job,
+)
+from awx.main.constants import ACTIVE_STATES
+from awx.main.dispatch.publish import task
+from awx.main.dispatch import get_local_queuename, reaper
+from awx.main.utils.common import (
+    ignore_inventory_computed_fields,
+    ignore_inventory_group_removal,
+    schedule_task_manager,
+)
+
+from awx.main.utils.external_logging import reconfigure_rsyslog
+from awx.main.utils.reload import stop_local_services
+from awx.main.utils.pglock import advisory_lock
+from awx.main.tasks.receptor import get_receptor_ctl, worker_info, worker_cleanup, administrative_workunit_reaper
+from awx.main.consumers import emit_channel_notification
+from awx.main import analytics
+from awx.conf import settings_registry
+from awx.main.analytics.subsystem_metrics import Metrics
+
+from rest_framework.exceptions import PermissionDenied
+
+logger = logging.getLogger('awx.main.tasks.system')
+
+__all__ = [
+    'RunJob',
+    'RunSystemJob',
+    'RunProjectUpdate',
+    'RunInventoryUpdate',
+    'RunAdHocCommand',
+    'handle_work_error',
+    'handle_work_success',
+    'apply_cluster_membership_policies',
+    'update_inventory_computed_fields',
+    'update_host_smart_inventory_memberships',
+    'send_notifications',
+    'purge_old_stdout_files',
+]
+
+HIDDEN_PASSWORD = '**********'
```

Comment:
This variable `HIDDEN_PASSWORD` isn't used anywhere in this file. Obviously we should not keep it.

But maybe this hints that we need stricter linter rules to guide a transition like this. We are using `black` as a replacement of these rules, but clearly some things are being missed. I would like to see some experimenting with resetting linter rules, and you could just use that to discover more rules we should put back in.

I mean, we whitelist rules right now:

https://github.com/ansible/awx/blob/d33a0d5ddeefc534e5c70f2bc95746479002cd17/tox.ini#L16

If you delete these, and re-run the linter, that should tell you more things. If anything it tells you is genuinely good feedback, then we should consider adding those rules to the allow list.